wurzel/steps/step_docling/docling_step.py

```diff
@@ -0,0 +1,133 @@
+# SPDX-FileCopyrightText: 2024 Deutsche Telekom AG
+#
+# SPDX-License-Identifier: CC0-1.0
+
+"""
+Note: Known Limitations with EasyOCR (`EasyOcrOptions`)
+
+1. Table structure is often lost or misaligned in the OCR output.
+2. Spelling inaccuracies are occasionally observed (e.g., "Verlängern" → "Verlängenng").
+3. URLs are not parsed correctly (e.g., "www.telekom.de/agb" → "www telekom delagb").
+
+While investigating EasyOCR issues and testing alternative OCR engines,
+we observed that some documents produced distorted text with irregular whitespace.
+This disrupts the natural sentence flow and significantly reduces readability.
+
+Example:
+"pra kti sche  i nform ati o nen zu  i h rer  fam i l y  card  basi c Li eber
+  Tel ekom   Kunde, schön,   dass  Si e  si ch  f ür..."
+
+Despite these limitations, we have decided to proceed with EasyOCR.
+"""
+
+import re
+from logging import getLogger
+
+from bs4 import BeautifulSoup
+from docling.datamodel.base_models import InputFormat
+from docling.datamodel.pipeline_options import EasyOcrOptions, PdfPipelineOptions
+from docling.document_converter import (
+    DocumentConverter,
+    PdfFormatOption,
+)
+from mistletoe import Document, HTMLRenderer
+
+from wurzel.datacontract.common import MarkdownDataContract
+from wurzel.step.typed_step import TypedStep
+
+from .settings import DoclingSettings
+
+log = getLogger(__name__)
+
+
+class CleanMarkdownRenderer(HTMLRenderer):
+    """
+    Custom Markdown renderer extending mistletoe's HTMLRenderer to clean up
+    unwanted elements from Markdown input.
+    """
+
+    @staticmethod
+    def render_html_block(token):
+        # Remove comments like <!-- image -->
+        cleaned = re.sub(r"<!--.*?-->", "", token.content)
+        return cleaned.strip()
+
+
+class DoclingStep(TypedStep[DoclingSettings, None, list[MarkdownDataContract]]):
+    """Step to return local Markdown files with enhanced PDF extraction for German."""
+
+    def __init__(self):
+        super().__init__()
+        self.converter = self.create_converter()
+
+    def create_converter(self) -> DocumentConverter:
+        """Create and configure the document converter for PDF and DOCX.
+
+        Returns:
+            DocumentConverter: Configured document converter.
+        """
+        pipeline_options = PdfPipelineOptions()
+        ocr_options = EasyOcrOptions(
+            force_full_page_ocr=self.settings.FORCE_FULL_PAGE_OCR
+        )
+        pipeline_options.ocr_options = ocr_options
+
+        return DocumentConverter(
+            allowed_formats=self.settings.FORMATS,
+            format_options={
+                InputFormat.PDF: PdfFormatOption(
+                    pipeline_options=pipeline_options,
+                )
+            },
+        )
+
+    @staticmethod
+    def clean_markdown_with_mistletoe(md_text: str) -> str:
```

Comment:
Incorrect return type
```suggestion
    def clean_markdown_with_mistletoe(md_text: str) -> str, str:
```